witness/src/Dscp/Witness/Mempool/Logic.hs

```diff
@@ -44,10 +45,13 @@ addTxToMempool
     .  MempoolCtx ctx m
     => MempoolVar
     -> GTxWitnessed
-    -> m ()
-addTxToMempool (Mempool pool conf, lock) tx =
-    writeToMempool @ctx pool lock
-        $ Pool.processTxAndInsertToMempool conf tx
+    -> m Bool
+addTxToMempool (Mempool pool conf, lock) tx = do
+    writeToMempool @ctx pool lock $ do
```

Comment:
```
writeToMempool @ctx pool lock $ 
  True <$ Pool.processTxAndInsertToMempool conf tx `catchError` (\_ -> pure False)
```

Reply:
This looks more like obfuscation, then style improvement.

Reply:
Dunno, maybe:
```
writeToMempool @ctx pool lock $
    True <$ Pool.processTxAndInsertToMempool conf tx
  `catchError`
    \_ -> pure False
```
?
:man_shrugging: :woman_shrugging: :man_shrugging:

Reply:
Frankly, everything looks bad, but @Heimdell's variant is easier to read.